==2711== Memcheck, a memory error detector
==2711== Copyright (C) 2002-2013, and GNU GPL'd, by Julian Seward et al.
==2711== Using Valgrind-3.10.1 and LibVEX; rerun with -h for copyright info
==2711== Command: ./uriProc
==2711== 
==2711== Invalid write of size 1
==2711==    at 0x4055BE: _up_parse_host(uri_proc_struct*, int) (uri_proc.c:314)
==2711==    by 0x405CA9: _up_parse(uri_proc_struct*, char*, int) (uri_proc.c:500)
==2711==    by 0x405E7F: up_create(char*, int, void*) (uri_proc.c:592)
==2711==    by 0x40679F: uri_proc_up_create_Test::TestBody() (UnitTest_uri_proc.c:88)
==2711==    by 0x42EA41: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x428A0D: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40E640: testing::Test::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40EECC: testing::TestInfo::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40F55D: testing::TestCase::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x415D08: testing::internal::UnitTestImpl::RunAllTests() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x42FE37: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x429773: bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==  Address 0x5d03d3b is 20 bytes after a block of size 7 alloc'd
==2711==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2711==    by 0x405587: _up_parse_host(uri_proc_struct*, int) (uri_proc.c:312)
==2711==    by 0x405CA9: _up_parse(uri_proc_struct*, char*, int) (uri_proc.c:500)
==2711==    by 0x405E7F: up_create(char*, int, void*) (uri_proc.c:592)
==2711==    by 0x40679F: uri_proc_up_create_Test::TestBody() (UnitTest_uri_proc.c:88)
==2711==    by 0x42EA41: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x428A0D: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40E640: testing::Test::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40EECC: testing::TestInfo::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40F55D: testing::TestCase::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x415D08: testing::internal::UnitTestImpl::RunAllTests() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x42FE37: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711== 
==2711== Invalid write of size 1
==2711==    at 0x40572B: _up_parse_host(uri_proc_struct*, int) (uri_proc.c:346)
==2711==    by 0x405CA9: _up_parse(uri_proc_struct*, char*, int) (uri_proc.c:500)
==2711==    by 0x405E7F: up_create(char*, int, void*) (uri_proc.c:592)
==2711==    by 0x40679F: uri_proc_up_create_Test::TestBody() (UnitTest_uri_proc.c:88)
==2711==    by 0x42EA41: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x428A0D: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40E640: testing::Test::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40EECC: testing::TestInfo::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40F55D: testing::TestCase::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x415D08: testing::internal::UnitTestImpl::RunAllTests() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x42FE37: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x429773: bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==  Address 0x5d066a2 is 0 bytes after a block of size 2 alloc'd
==2711==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2711==    by 0x4056B9: _up_parse_host(uri_proc_struct*, int) (uri_proc.c:339)
==2711==    by 0x405CA9: _up_parse(uri_proc_struct*, char*, int) (uri_proc.c:500)
==2711==    by 0x405E7F: up_create(char*, int, void*) (uri_proc.c:592)
==2711==    by 0x40679F: uri_proc_up_create_Test::TestBody() (UnitTest_uri_proc.c:88)
==2711==    by 0x42EA41: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x428A0D: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40E640: testing::Test::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40EECC: testing::TestInfo::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40F55D: testing::TestCase::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x415D08: testing::internal::UnitTestImpl::RunAllTests() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x42FE37: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711== 
==2711== Invalid read of size 1
==2711==    at 0x55C9943: vfprintf (vfprintf.c:1661)
==2711==    by 0x55D23D8: printf (printf.c:33)
==2711==    by 0x40642A: up_show(uri_proc_struct*) (uri_proc.c:792)
==2711==    by 0x406889: uri_proc_up_create_Test::TestBody() (UnitTest_uri_proc.c:91)
==2711==    by 0x42EA41: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x428A0D: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40E640: testing::Test::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40EECC: testing::TestInfo::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40F55D: testing::TestCase::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x415D08: testing::internal::UnitTestImpl::RunAllTests() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x42FE37: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x429773: bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==  Address 0x5d066a2 is 0 bytes after a block of size 2 alloc'd
==2711==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2711==    by 0x4056B9: _up_parse_host(uri_proc_struct*, int) (uri_proc.c:339)
==2711==    by 0x405CA9: _up_parse(uri_proc_struct*, char*, int) (uri_proc.c:500)
==2711==    by 0x405E7F: up_create(char*, int, void*) (uri_proc.c:592)
==2711==    by 0x40679F: uri_proc_up_create_Test::TestBody() (UnitTest_uri_proc.c:88)
==2711==    by 0x42EA41: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x428A0D: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40E640: testing::Test::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40EECC: testing::TestInfo::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40F55D: testing::TestCase::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x415D08: testing::internal::UnitTestImpl::RunAllTests() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x42FE37: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711== 
==2711== Conditional jump or move depends on uninitialised value(s)
==2711==    at 0x55C9943: vfprintf (vfprintf.c:1661)
==2711==    by 0x55D23D8: printf (printf.c:33)
==2711==    by 0x4063C4: up_show(uri_proc_struct*) (uri_proc.c:788)
==2711==    by 0x406A8F: uri_proc_up_create_from_host_Test::TestBody() (UnitTest_uri_proc.c:107)
==2711==    by 0x42EA41: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x428A0D: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40E640: testing::Test::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40EECC: testing::TestInfo::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40F55D: testing::TestCase::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x415D08: testing::internal::UnitTestImpl::RunAllTests() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x42FE37: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x429773: bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==  Uninitialised value was created by a heap allocation
==2711==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2711==    by 0x405F81: up_create_from_host(char*, int, void*) (uri_proc.c:627)
==2711==    by 0x4069A5: uri_proc_up_create_from_host_Test::TestBody() (UnitTest_uri_proc.c:104)
==2711==    by 0x42EA41: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x428A0D: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40E640: testing::Test::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40EECC: testing::TestInfo::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40F55D: testing::TestCase::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x415D08: testing::internal::UnitTestImpl::RunAllTests() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x42FE37: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x429773: bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x414992: testing::UnitTest::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711== 
==2711== Invalid write of size 1
==2711==    at 0x4055BE: _up_parse_host(uri_proc_struct*, int) (uri_proc.c:314)
==2711==    by 0x405FF2: up_create_from_host(char*, int, void*) (uri_proc.c:635)
==2711==    by 0x4069A5: uri_proc_up_create_from_host_Test::TestBody() (UnitTest_uri_proc.c:104)
==2711==    by 0x42EA41: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x428A0D: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40E640: testing::Test::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40EECC: testing::TestInfo::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40F55D: testing::TestCase::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x415D08: testing::internal::UnitTestImpl::RunAllTests() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x42FE37: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x429773: bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x414992: testing::UnitTest::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==  Address 0x5dda1e4 is 13 bytes after a block of size 7 alloc'd
==2711==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2711==    by 0x405587: _up_parse_host(uri_proc_struct*, int) (uri_proc.c:312)
==2711==    by 0x405FF2: up_create_from_host(char*, int, void*) (uri_proc.c:635)
==2711==    by 0x4069A5: uri_proc_up_create_from_host_Test::TestBody() (UnitTest_uri_proc.c:104)
==2711==    by 0x42EA41: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x428A0D: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40E640: testing::Test::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40EECC: testing::TestInfo::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x40F55D: testing::TestCase::Run() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x415D08: testing::internal::UnitTestImpl::RunAllTests() (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x42FE37: bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711==    by 0x429773: bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /home/montimage/workspace/montimage/uri_proc/unit_tests/uriProc)
==2711== 
==2711== 
==2711== HEAP SUMMARY:
==2711==     in use at exit: 72,704 bytes in 1 blocks
==2711==   total heap usage: 51,593 allocs, 51,592 frees, 1,292,233 bytes allocated
==2711== 
==2711== 72,704 bytes in 1 blocks are still reachable in loss record 1 of 1
==2711==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2711==    by 0x50BB62F: ??? (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.22)
==2711==    by 0x4010109: call_init.part.0 (dl-init.c:78)
==2711==    by 0x40101F2: call_init (dl-init.c:36)
==2711==    by 0x40101F2: _dl_init (dl-init.c:126)
==2711==    by 0x4001309: ??? (in /lib/x86_64-linux-gnu/ld-2.19.so)
==2711== 
==2711== LEAK SUMMARY:
==2711==    definitely lost: 0 bytes in 0 blocks
==2711==    indirectly lost: 0 bytes in 0 blocks
==2711==      possibly lost: 0 bytes in 0 blocks
==2711==    still reachable: 72,704 bytes in 1 blocks
==2711==         suppressed: 0 bytes in 0 blocks
==2711== 
==2711== For counts of detected and suppressed errors, rerun with: -v
==2711== ERROR SUMMARY: 6319 errors from 5 contexts (suppressed: 0 from 0)
